File: miniblog_front/src/server/dao/Blog.js
import Vue from "vue"

const BLOG_REST = "/miniblog/data/blog";
const BLOG_CATALOG = "/getBlogCatalog";
const BLOG_CATALOG_KEYWORD = "/getBlogCatalogByKeyword";

function Blog() {

    this.get = function (id) {
        return Vue.http.get(BLOG_REST + "/" + id);
    };
    this.getAll = function () {
        return Vue.http.get(BLOG_CATALOG);
    };
    this.put = function (blog) {
        return Vue.http.put(BLOG_REST + "/" + blog.id, this.normalBlog(blog));
    };

    this.delete = function (id) {
        return Vue.http.delete(BLOG_REST + "/" + id);
    };
    this.getBlogCatalogByKeyword = function (keyword) {
        return Vue.http.get(BLOG_CATALOG_KEYWORD, {params: {keyword: keyword}});
    };
    this.normalBlog = function (blog) {
        return {
            id: blog.id || null,
            title: blog.title || null,
            author: blog.author || null,
            tags: blog.tags || null,
            blogTags: blog.blogTags || null,
            previewContent: blog.previewContent || null,
            htmlContent: blog.htmlContent || null,
            updateTime: blog.updateTime || null,
            isShowBanner: blog.isShowBanner || null
        }
    }
}

export default Blog;

File: miniblog_back/src/main/java/red/medusa/miniblog/web/blog/bean/Blog.java
package red.medusa.miniblog.web.blog.bean;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.springframework.util.StringUtils;

import java.util.Date;

public class Blog {
    private String id;
    private String title;
    private String author;
    private String[] tags;
    private String blogTags;
    private String previewContent;
    private String htmlContent;
    @JsonFormat(timezone = "GMT+8", pattern = "yyyy-MM-dd HH:mm:ss")
    private Date updateTime;
    private boolean isShowBanner;


    public String getBlogTags() {
        return this.blogTags;
    }

    public String[] getTags() {
        return this.tags;
    }

    /**
     * 前台传入后端时设置blogTags
     */
    public void setTags(String[] tags) {
        if (this.blogTags == null)
            this.blogTags = StringUtils.arrayToCommaDelimitedString(tags);
        if (this.tags != null)
            return;
        this.tags = tags;
    }

    /**
     * 从数据库查询时设置tags
     */
    public void setBlogTags(String blogTags) {
        if (this.tags == null)
            this.tags = StringUtils.tokenizeToStringArray(blogTags, ",");
        if (this.blogTags != null)
            return;
        this.blogTags = blogTags;
    }


    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getAuthor() {
        return author;
    }

    public void setAuthor(String author) {
        this.author = author;
    }


    public String getPreviewContent() {
        return previewContent;
    }

    public void setPreviewContent(String previewContent) {
        this.previewContent = previewContent;
    }

    public String getHtmlContent() {
        return htmlContent;
    }

    public void setHtmlContent(String htmlContent) {
        this.htmlContent = htmlContent;
    }

    public Date getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(Date updateTime) {
        this.updateTime = updateTime;
    }

    public boolean getIsShowBanner() {
        return isShowBanner;
    }

    public void setIsShowBanner(boolean showBanner) {
        isShowBanner = showBanner;
    }
}
